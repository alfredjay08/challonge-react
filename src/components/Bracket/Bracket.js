import React from "react";
import { BracketContainer } from "../StyledComponents/StyledComponents";
import Match from "../Match";

const Bracket = ({
  matches,
  semi,
  isHighlighted = false,
  hovered,
  unhovered,
}) => {
  const activeMatches = matches && matches.length ? matches : [null, null];

  return (
    <BracketContainer semi={semi}>
      {matches &&
        activeMatches.map((match, index) => (
          <Match
            key={index}
            isHighlighted={isHighlighted}
            hovered={hovered}
            unhovered={unhovered}
            players={match?.players}
            matchId={match?._id}
          />
        ))}
    </BracketContainer>
  );
};

export default Bracket;
