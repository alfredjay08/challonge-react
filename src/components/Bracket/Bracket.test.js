import React from "react";
import renderer from "react-test-renderer";

import Bracket from "./Bracket";

it("renders when there are no props", () => {
  const tree = renderer.create(<Bracket />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders when there are only one match and no handler", () => {
  const match = {
    _id: 1001,
    players: [
      {
        data: {
          _id: 1001,
          name: "Derkila",
        },
        score: 3,
        winner: true,
      },
      {
        data: {
          _id: 1001,
          name: "Derkila",
        },
        score: 2,
        winner: false,
      },
    ],
  };
  const tree = renderer.create(<Bracket matches={match} />).toJSON();

  expect(tree).toMatchSnapshot();
});
