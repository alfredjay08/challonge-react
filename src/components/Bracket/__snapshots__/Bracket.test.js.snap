// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no props 1`] = `
<div
  className="sc-hLBbgP cKuqie"
/>
`;

exports[`renders when there are only one match and no handler 1`] = `
<div
  className="sc-hLBbgP cKuqie"
>
  <div
    className="sc-eDvSVe LqqaX"
  >
    <div
      className="sc-jSUZER hFnVAg"
      onMouseOver={[Function]}
    >
      <div
        className="sc-gKPRtg jAzlAk"
      >
        <img
          alt="Player Icon"
          src="/uploads/ninja-yellow.jpg"
          style={
            Object {
              "borderRadius": "50%",
              "width": "20px",
            }
          }
        />
      </div>
      <div
        className="sc-gKPRtg fOrJSf"
      />
      <div
        className="sc-gKPRtg jLsyYQ"
      >
        <span />
      </div>
    </div>
    <div
      className="sc-jSUZER hFnVAg"
      onMouseOver={[Function]}
    >
      <div
        className="sc-gKPRtg jAzlAk"
      >
        <img
          alt="Player Icon"
          src="/uploads/ninja-yellow.jpg"
          style={
            Object {
              "borderRadius": "50%",
              "width": "20px",
            }
          }
        />
      </div>
      <div
        className="sc-gKPRtg fOrJSf"
      />
      <div
        className="sc-gKPRtg jLsyYQ"
      >
        <span />
      </div>
    </div>
  </div>
  <div
    className="sc-eDvSVe LqqaX"
  >
    <div
      className="sc-jSUZER hFnVAg"
      onMouseOver={[Function]}
    >
      <div
        className="sc-gKPRtg jAzlAk"
      >
        <img
          alt="Player Icon"
          src="/uploads/ninja-yellow.jpg"
          style={
            Object {
              "borderRadius": "50%",
              "width": "20px",
            }
          }
        />
      </div>
      <div
        className="sc-gKPRtg fOrJSf"
      />
      <div
        className="sc-gKPRtg jLsyYQ"
      >
        <span />
      </div>
    </div>
    <div
      className="sc-jSUZER hFnVAg"
      onMouseOver={[Function]}
    >
      <div
        className="sc-gKPRtg jAzlAk"
      >
        <img
          alt="Player Icon"
          src="/uploads/ninja-yellow.jpg"
          style={
            Object {
              "borderRadius": "50%",
              "width": "20px",
            }
          }
        />
      </div>
      <div
        className="sc-gKPRtg fOrJSf"
      />
      <div
        className="sc-gKPRtg jLsyYQ"
      >
        <span />
      </div>
    </div>
  </div>
</div>
`;
